plumbing/api.go

```diff
@@ -104,6 +105,11 @@ func (api *API) ChainLs(ctx context.Context) <-chan interface{} {
 	return api.chain.Ls(ctx)
 }
 
+// ChainLatestState returns the latest state tree from the chain reader
+func (api *API) ChainLatestState(ctx context.Context) (state.Tree, error) {
```

Comment:
BLOCKING: The state.Tree is a read write interface to the data store. It's not an appropriate return value from the plumbing API. Better to implement a `GetActorFromLatestState`. This will simplify testing.

Reply:
but getting the latest tree is sometimes what you want, isn't it? does it make sense to have a read only interface that is defined here?

Reply:
@phritz A read only interface would expose `GetActor` and possibly `ForEachActor` I don't really have a problem with it so long as it's read-only, but it seems a little clunky to nest the API this way. It's:
``` go
api.ChainGetActor(ctx, addr)
```
vs.
``` go
api.ChainLatestState(ctx).GetActor(addr)
```

Reply:
Coming at it from another angle, say you are a developer trying to inspect an actor. As you peruse the API, you'd have to be pretty familiar with the system to understand that you should start with chain and then get latest state which would then expose the GetActor method. 

The plumbing API is not supposed to be easy to use, but where we have the option to make it easier at no real cost, we should take it.